lib/Installation/ModuleRegistration/ModuleRegCodeController.pm

```diff
@@ -59,4 +61,26 @@ sub trust_gnupg_key {
     $self->get_untrusted_GPG_popup()->press_trust();
 }
 
+sub get_module_ha_regcode_page {
+    my ($self) = @_;
+    die "Extension and Module Registration Codes page" unless $self->{ModuleHARegCodePage}->is_shown();
+    return $self->{ModuleHARegCodePage};
+}
+
+sub wait_ha_regcode_page {
+    my ($self, $args) = @_;
+    YuiRestClient::Wait::wait_until(object => sub {
+            $self->{ModuleHARegCodePage}->is_shown({timeout => 0});
+    }, %$args);
+}
+
+sub add_ha_registration_code {
+    my ($self, $regcode, $timeout) = @_;
+    $self->wait_ha_regcode_page({timeout => $timeout, interval => 2,
```

Comment:
once you put this out as commented in the test layer, the method come back to be simple, we should try to avoid too complex method (if we would do unit testing against our own framework, we would notice easily this, but as we don't we can just spot it with some luck.